EMCAL/EMCALrec/AliEMCALClusterizerv3.h

```diff
@@ -0,0 +1,122 @@
+#ifndef ALIEMCALCLUSTERIZERV3_H
+#define ALIEMCALCLUSTERIZERV3_H
+
+/* Copyright(c) 1998-1999, ALICE Experiment at CERN, All rights reserved. *
+ * See cxx source for full Copyright notice                               */
+
+
+#include "AliEMCALClusterizerv1.h"
+class AliEMCALRecPoint; 
+class AliEMCALDigit;
+
+// Define numbers rows/columns for topological representation of cells
+const Int_t kNrows    = (24+1)*(6+4);  // 10x supermodule rows (6 for EMCAL, 4 for DCAL). +1 accounts for topological gap between two supermodules
+const Int_t kNcolumns = 48*2+1;        // 2x  supermodule columns + 1 empty space in between for DCAL (not used for EMCAL)
+
+struct cellWithE {
+  cellWithE() : energy(0.), row(0), column(0) {}
+  cellWithE(Float_t e, Int_t r, Int_t c) : energy(e), row(r), column(c) {}
+  // std::sort will require operator< to compile.
+  bool operator<( cellWithE const& rhs ) const
+     { return energy < rhs.energy; }
+  Float_t energy;
+  Int_t row;
+  Int_t column;
+};
+
+
+//_________________________________________________________________________
+/// \class AliEMCALClusterFinder
+/// \ingroup EMCALrec
+/// \brief Meta class for recursive clusterizer
+///
+///  Implementation of same algorithm version as in AliEMCALClusterizerv2,
+///  but optimized.
+///
+/// \author Rudiger Haake (Yale)
+//_________________________________________________________________________
+class AliEMCALClusterFinder: public TObject
+{
+  public:
+    AliEMCALClusterFinder(TObjArray* outputArray, AliEMCALGeometry* geometry, Double_t timeCut, Double_t timeMin, Double_t timeMax, Double_t gradientCut, Bool_t doEnergyGradientCut, Double_t thresholdSeedE, Double_t thresholdCellE);
+    ~AliEMCALClusterFinder();
+
+    Int_t               FindClusters(TClonesArray* digits);
+    TObjArray*          GetFoundClusters() {return fFoundClusters;}
+    
+  private:
+    AliEMCALRecPoint*   GetClusterFromNeighbours(AliEMCALRecPoint* recPoint, Int_t row, Int_t column);
+    void                GetTopologicalRowColumn(AliEMCALDigit* digit, Int_t& row, Int_t& column);
+
+    AliEMCALGeometry*   fEMCALGeometry;
+    cellWithE           fSeedList[kNrows*kNcolumns];
+    AliEMCALDigit*      fDigitMap[kNrows][kNcolumns];
+    Bool_t              fCellMask[kNrows][kNcolumns];
```

Comment:
The three arrays are working arrays which I think should be transient (adding "//!<!")

Reply:
Yes, and it seems I forget to give a basic documentation for the class member. Added.